gitea/giteaclient.go

```diff
@@ -0,0 +1,371 @@
+/*
+Copyright 2020 The Flux CD contributors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package gitea
+
+import (
+	"context"
+	"fmt"
+
+	"code.gitea.io/sdk/gitea"
+	"github.com/fluxcd/go-git-providers/gitprovider"
+)
+
+// giteaClientImpl is a wrapper around *gitea.Client, which implements higher-level methods,
+// operating on the gitea structs. TODO: Verify pagination is implemented for all List* methods,
+// all returned objects are validated, and HTTP errors are handled/wrapped using handleHTTPError.
+// This interface is also fakeable, in order to unit-test the client.
+type giteaClient interface {
```

Comment:
We are luckily no one rely on this interface yet. Normally I wouldn't advise for such a large interface, in GO you should try to keep interfaces small, and composable. I get it you want to keep existing code structure, nevertheless, please modify the functions structures as needed (e.g. some of them don't need `context.Context` parameters)